pkg/store/cache/cache.go

```diff
@@ -0,0 +1,335 @@
+	}, []string{"item_type"})
+	c.evicted.WithLabelValues(cacheTypePostings)
+	c.evicted.WithLabelValues(cacheTypeSeries)
+
+	c.added = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name: "thanos_store_index_cache_items_added_total",
+		Help: "Total number of items that were added to the index cache.",
+	}, []string{"item_type"})
+	c.added.WithLabelValues(cacheTypePostings)
+	c.added.WithLabelValues(cacheTypeSeries)
+
+	c.requests = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name: "thanos_store_index_cache_requests_total",
+		Help: "Total number of requests to the cache.",
+	}, []string{"item_type"})
+	c.requests.WithLabelValues(cacheTypePostings)
+	c.requests.WithLabelValues(cacheTypeSeries)
+
+	c.overflow = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name: "thanos_store_index_cache_items_overflowed_total",
+		Help: "Total number of items that could not be added to the cache due to being too big.",
+	}, []string{"item_type"})
+	c.overflow.WithLabelValues(cacheTypePostings)
+	c.overflow.WithLabelValues(cacheTypeSeries)
+
+	c.hits = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name: "thanos_store_index_cache_hits_total",
+		Help: "Total number of requests to the cache that were a hit.",
+	}, []string{"item_type"})
+	c.hits.WithLabelValues(cacheTypePostings)
+	c.hits.WithLabelValues(cacheTypeSeries)
+
+	c.current = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "thanos_store_index_cache_items",
+		Help: "Current number of items in the index cache.",
+	}, []string{"item_type"})
+	c.current.WithLabelValues(cacheTypePostings)
+	c.current.WithLabelValues(cacheTypeSeries)
+
+	c.currentSize = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Name: "thanos_store_index_cache_items_size_bytes",
+		Help: "Current byte size of items in the index cache.",
+	}, []string{"item_type"})
+	c.currentSize.WithLabelValues(cacheTypePostings)
+	c.currentSize.WithLabelValues(cacheTypeSeries)
+
+	c.cancelled = prometheus.NewCounterVec(prometheus.CounterOpts{
+		Name: "thanos_store_index_cache_set_cancels_total",
+		Help: "Number of requests that timed out or cancelled before being set to cache. Might indicate slow cache",
+	}, []string{"item_type"})
+	c.cancelled.WithLabelValues(cacheTypePostings)
+	c.cancelled.WithLabelValues(cacheTypeSeries)
+
+	if reg != nil {
+		reg.MustRegister(prometheus.NewGaugeFunc(prometheus.GaugeOpts{
+			Name: "thanos_store_index_cache_max_size_bytes",
+			Help: "Maximum number of bytes to be held in the index cache.",
+		}, func() float64 {
+			return float64(c.maxSizeBytes)
+		}))
+		reg.MustRegister(prometheus.NewGaugeFunc(prometheus.GaugeOpts{
+			Name: "thanos_store_index_cache_max_item_size_bytes",
+			Help: "Maximum number of bytes for single entry to be held in the index cache.",
+		}, func() float64 {
+			return float64(c.maxItemSizeBytes)
+		}))
+		reg.MustRegister(c.requests, c.hits, c.added, c.evicted, c.current, c.currentSize, c.overflow, c.cancelled)
+	}
+
+	if opts.MaxSizeBytes > 1e12 {
```

Comment:
Isn't the `1e12` the maximum number of _items_ in the LRU, unrelated to their size?

Reply:
You are right, changing the comment (and removing this check)